### PR TITLE
[DeviceMesh] Make our mesh_dim kwarg naming consistent

### PR DESCRIPTION
**Pull request by wz337**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114707

Changing size(self, dim: Optional[int] = None) to def size(self, mesh_dim: Optional[int] = None) so it is consistent with the rest of our APIs.

We also update this API usage change in both PT and internal (pyper, APS).

Differential Revision: [D51602986](https://our.internmc.facebook.com/intern/diff/D51602986/)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @tianyu-l @wconstab @yf225 @kiukchung @d4l3k @lucasllc

### CODE REVIEW

**Review by XilunWu — APPROVED** [review]
thx for making the arg naming more consistent!

**Review by wanchaol — APPROVED** [review]

File: torch/distributed/_device_mesh.py [hunk 1]
```diff
@@ -354,8 +354,8 @@ def get_dim_groups(
                 )
             return dim_groups
 
-    def size(self, dim: Optional[int] = None) -> int:
-        return self.mesh.numel() if dim is None else self.mesh.size(dim)
+    def size(self, mesh_dim: Optional[int] = None) -> int:
+        return self.mesh.numel() if mesh_dim is None else self.mesh.size(mesh_dim)
```
Comment (wanchaol) [inline torch/distributed/_device_mesh.py:358]: do we want to add documentation in this PR or in follow up PR?

Comment (wz337) [reply torch/distributed/_device_mesh.py:358]: > do we want to add documentation in this PR or in follow up PR?

I included all the documentation in the last one where we make device_mesh public.

**Review by fegin — APPROVED** [review]
There is a fsdp/tp compilation test failure. Probably not related, but it is good to check.

**Pull request closed without merging**